examples/next-routes/next.config.js

```diff
@@ -0,0 +1,20 @@
+const { routes } = require('./routes')
+
+module.exports = {
+  experimental: {
+    rewrites() {
+      const rewrites = []
```

Comment:
I feels like this should just be the exported by the package. If folks want to customize it, they can roll it on their own, but otherwise, `withRoutes(routes)` or `withRoutes(/* define it all inline */)`.

Reply:
The package (next-routes) is unmaintained (wasn't created by us) and we want people to incrementally move away from it as it has a lot of pitfalls like a completely different routing API with non-standard naming, sending a manifest of all routes to the client etc.

Reply:
It needs to yet improve a lot and as @jaredpalmer said, it should be implemented by `next/router`.
For this new use case, it may need to implement a new Composed link if `route` property is mentioned.
I'm using Next.js link component implementation:

```js
/* eslint-disable jsx-a11y/anchor-has-content */
import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { withRouter } from "next/router";
import NextLink from "next/link";
import MuiLink from "@material-ui/core/Link";

const NextComposed = React.forwardRef((props, ref) => {
  const { as, href, prefetch, ...other } = props;

  return (
    <NextLink href={href} prefetch={prefetch} as={as}>
      <a ref={ref} {...other} />
    </NextLink>
  );
});

NextComposed.propTypes = {
  as: PropTypes.string,
  href: PropTypes.string,
  prefetch: PropTypes.bool
};

// A styled version of the Next.js Link component:
// https://next.js.org/docs/#with-link
const Link = (props) => {
 ....
 route,
} = props;


  const className = clsx(classNameProps, {
    [activeClassName]: router.pathname === props.href && activeClassName
  });

  if (naked) {
    return <NextComposed className={className} ref={innerRef} {...other} />
  }

  // Here we'll need to add a new if statement to check if `route` is mentioned by props
  if (route) {
    return the custom NextComposed by route prop instead "as"
  }

  return <MuiLink component={NextComposed} className={className} ref={innerRef} {...other} />
};
```

Reply:
The Link implementation doesn't fits right.

```js
<Link route="/blog/:dynamic/first">
  <a>To /blog/:dynamic/first!</a>
</Link>
```

It gets the `incompatible-href-as` error.